static/js/compose.js

```diff
@@ -250,6 +251,9 @@ export function send_message(request = create_message_object()) {
     }
 
     transmit.send_message(request, success, error);
+    if (request.type === "private" && $(".expanded-private-messages").children().length === 0) {
+        pm_list.append_new_active_pm_in_collapsed_pms();
+    }
```

Comment:
This is the wrong place to add this, because messages can be sent on other clients. This belongs in the `server_events_dispatch.js` code path, likely in whatever "update the pm_list" function is called when we receive a new message from the server (including ones we sent).

You can test for this sort of thing easily by using two browsers in your development environment to send messages.

BTW the new application feature tutorial is worth reading, as it explains this category of bug and why it's important to avoid it (especially https://zulip.readthedocs.io/en/latest/tutorials/new-feature-tutorial.html#update-the-frontend)

Reply:
Thanks Tim, Yes, this was not needed in code, there was some other bug which i handled with #21726.